src/herder/HerderImpl.cpp

```diff
@@ -767,19 +767,27 @@ HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger)
                               lcl.header.scpValue);
 }
 
+bool
+HerderImpl::timeForUpgrade() const
```

Comment:
I don't think you need to add this helper: it's probably much simpler to change the way PREFERRED_UPGRADE_DATETIME is done by configuring it by default to have the value of epoch (instead of using an optional field)